src/Compilers/Core/Portable/TreeDumper.cs

```diff
@@ -166,8 +166,13 @@ private void DoDumpXML(TreeDumperNode node, string indent, string relativeIndent
         private static bool IsDefaultImmutableArray(Object o)
         {
             var ti = o.GetType().GetTypeInfo();
-            return ti.IsGenericType && ti.GetGenericTypeDefinition() == typeof(ImmutableArray<>) &&
-                (bool)ti.GetDeclaredMethod("get_IsDefault").Invoke(o, Array.Empty<object>());
+            if (ti.IsGenericType && ti.GetGenericTypeDefinition() == typeof(ImmutableArray<>))
+            {
+                var result = ti?.GetDeclaredMethod("get_IsDefault")?.Invoke(o, Array.Empty<object>());
+                return result is bool b && b;
```

Comment:
`return ti?.GetDeclaredMethod(...).Invoke(...) == true;`

Reply:
That doesn't work here because the return of `Invoke` is `object` and we can't compare `object == bool`. Know this because I had the exact same thought here :smile:

Reply:
Then perhaps:
`true.Equals(ti?.GetDeclaredMethod(...).Invoke(...))`

---
In reply to: [376560180](https://github.com/dotnet/roslyn/pull/41485#discussion_r376560180) [](ancestors = 376560180)